### PR TITLE
Broke down ReferenceFrame.orient() into five simpler methods.

### LINKED ISSUE #17758: Break sympy.physics.vector.frame.RefererenceFrame.orient() into 5 seperate methods.
`sympy.physics.vector.frame.RefererenceFrame.orient()` is a very long, complicated, and unwieldy method and there are just two many different types of arguments that it accepts, which requires very detailed explanations. It should be broken up into sub-methods. These could be:

`ReferenceFrame.orient_body()`
`orient_axis()`
`orient_dcm()`
etc.

### PR DESCRIPTION
**Pull request by sksingh1202**
#### References to other Issues or PRs
Fixes #17758 

#### Brief description of what is fixed or changed
Five new methods, one for each rot_type have been introduced instead of `sympy.physics.vector.frame.RefererenceFrame.orient()`, to simplify it. Also, the methods which previously used ReferenceFrame.orient() method have been modified accordingly.

#### Other comments
The new methods are orient_<rot_type>() where <rot_type> is 'axis', 'dcm', 'body', 'space' and 'quaternion' for each method which involves the relevant rotation type.

#### Release Notes

* physics.vector
    * Broke down ReferenceFrame.orient() into five simpler methods.
    * Modified the methods which earlier used ReferenceFrame.orient() method so that they use the new methods

### CODE REVIEW

File: sympy/matrices/tests/test_commonmatrix.py [hunk 1]
```diff
@@ -466,19 +466,6 @@ def test_conjugate():
...
-    a = OperationsOnlyMatrix([[Add(x, x, evaluate=False)]])
-    assert a[0] != 2*x
-    assert a.doit() == Matrix([[2*x]])
-
-
-def test_evalf():
-    a = OperationsOnlyMatrix(2, 1, [sqrt(5), 6])
-    assert all(a.evalf()[i] == a[i].evalf() for i in range(2))
-    assert all(a.evalf(2)[i] == a[i].evalf(2) for i in range(2))
-    assert all(a.n(2)[i] == a[i].n(2) for i in range(2))
```
Comment (moorepants) [inline sympy/matrices/tests/test_commonmatrix.py]: Tests should not be removed without a valid reason.

File: sympy/matrices/tests/test_matrices.py [hunk 2]
```diff
@@ -597,33 +597,6 @@ def test_issue_17247_expression_blowup_30():
...
-    M = Matrix([
-        [x + 1, 1 - x,     0,     0],
-        [1 - x, x + 1,     0, x + 1],
-        [    0, 1 - x, x + 1,     0],
-        [    0,     0,     0, x + 1]])
-    assert M.LUsolve(ones(4, 1)) == Matrix([
-        [(x + 1)/(4*x)],
-        [(x - 1)/(4*x)],
-        [(x + 1)/(4*x)],
-        [    1/(x + 1)]])
```
Comment (moorepants) [inline sympy/matrices/tests/test_matrices.py]: Don't remove.

File: sympy/matrices/tests/test_matrices.py [hunk 3]
```diff
@@ -643,29 +616,6 @@ def test_issue_16823():
...
-        ])
-    assert M.rref() == (Matrix([
-        [1, 0, 0, 0, 0, 0, 0, 0,  1/2],
-        [0, 1, 0, 0, 0, 0, 0, 0, -1/2],
-        [0, 0, 1, 0, 0, 0, 0, 0,  1/2],
-        [0, 0, 0, 1, 0, 0, 0, 0, -1/2],
-        [0, 0, 0, 0, 1, 0, 0, 0,    0],
-        [0, 0, 0, 0, 0, 1, 0, 0, -1/2],
-        [0, 0, 0, 0, 0, 0, 1, 0,    0],
-        [0, 0, 0, 0, 0, 0, 0, 1, -1/2]]), (0, 1, 2, 3, 4, 5, 6, 7))
```
Comment (moorepants) [inline sympy/matrices/tests/test_matrices.py]: Don't remove.

File: sympy/physics/vector/frame.py [hunk 4]
```diff
@@ -475,72 +475,40 @@ def dcm(self, otherframe):
...
         return outdcm
-
-    def orient(self, parent, rot_type, amounts, rot_order=''):
-        """Sets the orientation of this reference frame relative to another
-        (parent) reference frame.
-
-        Parameters
-        ==========
-
-        parent : ReferenceFrame
```
Comment (moorepants) [inline sympy/physics/vector/frame.py]: Don't remove any public methods. This method should call all of the new methods.

File: sympy/physics/vector/frame.py [hunk 5]
```diff
@@ -554,45 +522,182 @@ def orient(self, parent, rot_type, amounts, rot_order=''):
...
+        self._dcm_dict.update({parent: parent_orient_dcm.T})
+        parent._dcm_dict.update({self: parent_orient_dcm})
+        # Also update the dcm cache after resetting it
+        self._dcm_cache = {}
+        self._dcm_cache.update({parent: parent_orient_dcm.T})
+        parent._dcm_cache.update({self: parent_orient_dcm})
+        wvec = self._w_diff_dcm(parent)
+        self._ang_vel_dict.update({parent: wvec})
+        parent._ang_vel_dict.update({self: -wvec})
+        self._var_dict = {}
```
Comment (moorepants) [inline sympy/physics/vector/frame.py]: You seem to have a lot of repeated code in these methods. The repeated code should be in a single hidden method so there is no redundancy.

**Comment by moorepants** [comment]
My primary recommendation would be that you should write unit tests for the new methods first. Then work on the code changes.

File: sympy/matrices/tests/test_commonmatrix.py (hunk 1 above)
Comment (sksingh1202) [reply sympy/matrices/tests/test_commonmatrix.py]: I am sorry for that. It was by mistake and was to be a part of issue #18666. I would revert those two commits soon.

File: sympy/physics/vector/frame.py (hunk 5 above)
Comment (sksingh1202) [reply sympy/physics/vector/frame.py]: Thank you for that, I would certainly try making a new method to prevent the repetition of code.

**Comment by sksingh1202** [comment]
Sure, thank you for your recommendation. I just wanted to clarify that do you mean that I should be trying to add new test cases to `sympy.physics.vector.tests.test_frame.py` for the newly created methods?

**Comment by czgdp1807** [comment]
@sksingh1202 Would you like to complete this PR? Thanks for your contributions.

**Comment by czgdp1807** [comment]
Closing in favor of https://github.com/sympy/sympy/pull/20318

**Pull request closed without merging**